kymatio/scattering2d/backend/tensorflow_backend.py

```diff
@@ -16,11 +16,29 @@ def __init__(self, pad_size, input_size):
                 size of the original signal
         """
         self.pad_size = pad_size
+        self.input_size = input_size
 
     def __call__(self, x):
+        pad_size_tmp = list(self.pad_size)
+        # This handles the case where the padding is equal to the image size
+        if pad_size_tmp[0] == self.input_size[0]:
+            pad_size_tmp[0] -= 1
+            pad_size_tmp[1] -= 1
+        if pad_size_tmp[2] == self.input_size[1]:
+            pad_size_tmp[2] -= 1
+            pad_size_tmp[3] -= 1
+
         paddings = [[0, 0]] * len(x.shape[:-2])
-        paddings += [[self.pad_size[0], self.pad_size[1]], [self.pad_size[2], self.pad_size[3]]]
-        return tf.pad(x, paddings, mode="REFLECT")
+        paddings += [[pad_size_tmp[0], pad_size_tmp[1]], [pad_size_tmp[2], pad_size_tmp[3]]]
+
+        x_padded = tf.pad(x, paddings, mode="REFLECT")
+
+        pad_size_tmp = list(self.pad_size)
```

Comment:
Not really used except for the comparison below. Maybe just do a straight `self.pad_size[0] == self.input_size[0]` and `self.pad_size[2] == self.input_size[1]` comparison?

Reply:
we cant do `self.pad_size` as a list.  if we did, note that we edit it during the above computation and then check the size (which shouldn't change).